plugins/search/src/alpha.tsx

```diff
@@ -0,0 +1,485 @@
+import { useNavigate } from 'react-router-dom';
+
+import {
+  makeStyles,
+  Theme,
+  Grid,
+  Box,
+  Paper,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Button,
+  IconButton,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import SearchIcon from '@material-ui/icons/Search';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+
+import {
+  CatalogIcon,
+  Content,
+  DocsIcon,
+  Header,
+  Page,
+  useSidebarPinState,
+} from '@backstage/core-components';
+import {
+  createRouteRef,
+  useApi,
+  DiscoveryApi,
+  IdentityApi,
+  discoveryApiRef,
+  identityApiRef,
+} from '@backstage/core-plugin-api';
+
+import {
+  createPlugin,
+  createApiExtension,
+  createPageExtension,
+  createExtensionInput,
+  createNavItemExtension,
+  createExtensionDataRef,
+  createExtension,
+} from '@backstage/frontend-plugin-api';
+
+import {
+  catalogApiRef,
+  CATALOG_FILTER_EXISTS,
+} from '@backstage/plugin-catalog-react';
+
+import {
+  DefaultResultListItem,
+  SearchBar,
+  SearchFilter,
+  SearchPagination,
+  SearchResult as SearchResults,
+  SearchResultPager,
+  useSearch,
+  SearchContextProvider,
+} from '@backstage/plugin-search-react';
+import { SearchResult } from '@backstage/plugin-search-common';
+import { searchApiRef } from '@backstage/plugin-search-react';
+import { searchResultItemExtensionData } from '@backstage/plugin-search-react/alpha';
+
+import { SearchClient } from './apis';
+import { SearchType } from './components/SearchType';
+import { UrlUpdater } from './components/SearchPage/SearchPage';
+import { SidebarSearchModal } from './components/SidebarSearchModal/SidebarSearchModal';
+import { useSearchModal } from './components/SearchModal';
+
+/** @alpha */
+export const SearchApi = createApiExtension({
+  factory: {
+    api: searchApiRef,
+    deps: { discoveryApi: discoveryApiRef, identityApi: identityApiRef },
+    factory: ({
+      identityApi,
+      discoveryApi,
+    }: {
+      identityApi: IdentityApi;
+      discoveryApi: DiscoveryApi;
+    }) => new SearchClient({ discoveryApi, identityApi }),
+  },
+});
+
+const useSearchPageStyles = makeStyles((theme: Theme) => ({
+  filter: {
+    '& + &': {
+      marginTop: theme.spacing(2.5),
+    },
+  },
+  filters: {
+    padding: theme.spacing(2),
+    marginTop: theme.spacing(2),
+  },
+}));
+
+const searchRouteRef = createRouteRef({ id: 'plugin.search.page' });
+
+/** @alpha */
+export const SearchPage = createPageExtension({
+  id: 'plugin.search.page',
+  defaultPath: '/search',
+  routeRef: searchRouteRef,
+  inputs: {
+    items: createExtensionInput({
+      item: searchResultItemExtensionData,
+    }),
+  },
+  loader: async ({ inputs }) => {
+    const getResultItemComponent = (result: SearchResult) => {
+      const value = inputs.items.find(({ item }) => item?.predicate?.(result));
+      return value?.item.component ?? DefaultResultListItem;
+    };
+
+    const Component = () => {
```

Comment:
> [!Note]
This is the default `SearchPage` which was previously configured via code on the non-declarative integrated App ([packages/app/src/components/search/SearchPage](https://github.com/backstage/backstage/blob/master/packages/app/src/components/search/SearchPage.tsx)).